src/main/java/it/gov/pagopa/apiconfig/testingsupport/service/GenericQueryService.java

```diff
@@ -2,51 +2,103 @@
-import javax.persistence.Query;
-import org.hibernate.query.internal.NativeQueryImpl;
+import lombok.RequiredArgsConstructor;
+import org.hibernate.query.NativeQuery;
 import org.hibernate.transform.AliasToEntityMapResultTransformer;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.jdbc.BadSqlGrammarException;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.PersistenceException;
+import javax.persistence.Query;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 @Service
+@RequiredArgsConstructor
 public class GenericQueryService {
 
   @PersistenceContext
-  private EntityManager entityManager;
+  EntityManager entityManager;
+
   @Value("${dangerous.sql.keywords}")
-  private List<String> dangerousKeywordsList;
+  List<String> dangerousKeywordsList;
 
-  @Autowired
-  private JdbcTemplate jdbcTemplate;
+  private final JdbcTemplate jdbcTemplate;
 
   @Transactional
-  public List getQueryResponse(String query) {
+  public List<Object> getQueryResponse(String query) {
     String sanitizedQuery = sanitizeQuery(query);
     try {
-      Query nquery = entityManager.createNativeQuery(sanitizedQuery);
-      NativeQueryImpl nativeQuery = (NativeQueryImpl) nquery;
-      nativeQuery.setResultTransformer(AliasToEntityMapResultTransformer.INSTANCE);
-      if(modifying(query)){
-        return Collections.singletonList(nativeQuery.executeUpdate());
+      Collection<?> results = executeQuery(sanitizedQuery);
+      List<Object> objectList = new ArrayList<>(results.size());
+      objectList.addAll(results);
+      return objectList;
+    } catch (PersistenceException pExc) {
+      throw new AppException(AppError.WRONG_QUERY_GRAMMAR);
+    } catch (Exception e) {
+      throw new AppException(AppError.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  @Transactional
+  public List<Object> getMassiveQueryResponse(MultipartFile file) {
+    if (file.isEmpty()) {
+      throw new AppException(AppError.FILE_EMPTY);
+    }
+
+    List<String> queries = new ArrayList<>();
+    try {
+      try (BufferedReader reader = new BufferedReader(new InputStreamReader(file.getInputStream()))) {
+        StringBuilder queryBuilder = new StringBuilder();
+        String line;
+
+        while ((line = reader.readLine()) != null) {
+          line = line.trim();
+
+          // Skip empty lines and SQL comments
+          if (line.isEmpty() || line.startsWith("--") || line.startsWith("//") || line.startsWith("/*")) {
```

Comment:
A possible comment could be the following:
```
/* USEFUL QUERY:
 * UPDATE TABLE X SET ....
*/
```
With the highlighted code line, the exclusion of line ` * UPDATE TABLE X SET ....` is not performed, so it is executed (and maybe the execution fails because of `*` character).
Maybe could be useful to include also `*` and `*\` in this line?